web/modules/custom/dpl_url_proxy/src/Form/ProxyUrlConfigurationForm.php

```diff
@@ -0,0 +1,303 @@
+      ];
+      $form['hostnames'][$index]['hostname'] = [
+        '#type' => 'textfield',
+        '#title' => $this->t('Hostname', [], $t_opts),
+        '#default_value' => $saved_values['hostnames'][$index]['hostname'] ?? '',
+        '#required' => TRUE,
+      ];
+
+      $form['hostnames'][$index]['expression'] = [
+        '#type' => 'details',
+        '#title' => $this->t(
+          'Replacement %configured',
+          [
+            '%configured' =>
+            $this->regexIsConfiguredLabel($saved_values['hostnames'][$index] ?? []),
+          ],
+          $t_opts
+        ),
+      ];
+
+      $form['hostnames'][$index]['expression']['regex'] = [
+        '#type' => 'textfield',
+        '#title' => $this->t('Regular expression'),
+        '#size' => 30,
+        '#default_value' => $saved_values['hostnames'][$index]['expression']['regex'] ?? '',
+        '#description' => $this->t(
+          'Use regular expression to substitut parts of the url, e.g. "<em>%regex</em>".',
+          ['%regex' => '/bib\w{5,6}/'],
+          $t_opts
+        ),
+      ];
+
+      $form['hostnames'][$index]['expression']['replacement'] = [
+        '#type' => 'textfield',
+        '#title' => $this->t('Replacement'),
+        '#size' => 30,
+        '#default_value' => $saved_values['hostnames'][$index]['expression']['replacement'] ?? '',
+        '#description' => $this->t('The replacement value for the regular expression.', [], $t_opts),
+      ];
+
+      $form['hostnames'][$index]['disable_prefix'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Do not use proxy prefix for this hostname', [], $t_opts),
+        '#default_value' => isset($saved_values['hostnames'][$index]['disable_prefix'])
+        ? $saved_values['hostnames'][$index]['disable_prefix']
+        : FALSE,
+      ];
+
+      $form['hostnames'][$index]['remove_this'] = [
+        '#name' => $index,
+        '#type' => 'submit',
+        '#value' => $this->t('Remove this', [], $t_opts),
+        '#submit' => ['::removeOne'],
+        '#limit_validation_errors' => [],
+        '#ajax' => [
+          'callback' => '::addmoreCallback',
+          'wrapper' => 'hostnames-fieldset-wrapper',
+        ],
+      ];
+    }
+
+    $form['hostnames']['actions'] = [
+      '#type' => 'actions',
+    ];
+    $form['hostnames']['actions']['add'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Add one more', [], $t_opts),
+      '#submit' => ['::addOne'],
+      '#limit_validation_errors' => [],
+      '#ajax' => [
+        'callback' => '::addmoreCallback',
+        'wrapper' => 'hostnames-fieldset-wrapper',
+      ],
+    ];
+
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Submit', [], $t_opts),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'form_api_example_ajax_addmore';
+  }
```

Comment:
Please correct the form id.

`form_api_example_ajax_addmore` does not seem right.